packages/dapp-svelte-wallet/api/src/lib-wallet.js

```diff
@@ -1146,8 +1195,16 @@ export function makeWallet({
    */
   const updatePaymentRecord = ({ actions, ...preDisplay }) => {
     const displayPayment = fillInSlots(dehydrate(harden(preDisplay)));
-    const paymentRecord = { ...preDisplay, actions, displayPayment };
-    payments.set(paymentRecord.payment, harden(paymentRecord));
+    const paymentRecord = addMeta({
+      ...preDisplay,
+      actions,
+      displayPayment,
+    });
+    if (payments.has(paymentRecord.payment)) {
```

Comment:
Is this required by the metadata change, or unrelated?

Reply:
Unrelated, but a bug.  Without this, the payment record setting could throw.